File: backend/StudentBlogAPI/Model/DTOs/PostDtos.cs
using StudentBlogAPI.Model.Entities;

namespace StudentBlogAPI.Model.DTOs;

public record CreatePostReqDto(
    string Title,
    string Content
);

public record UpdatePostReqDto(
    string Title,
    string Content,
    PublicationStatus Status
);

public record PostResDto(
    int Id,
    int UserId,
    string Title,
    string Content,
    PublicationStatus Status,
    DateTime CreatedAt,
    DateTime UpdatedAt
);

File: frontend/types/converted-dtos/PostDtos.ts
import { PublicationStatus } from "../internal-enums";

export type CreatePostReqDto = {
  title: string;
  content: string;
};
export type UpdatePostReqDto = {
  title: string;
  content: string;
  status: PublicationStatus; // PublicationStatus is an enum, refer to the enum definition for possible values
};
export type PostResDto = {
  id: number;
  userId: number;
  title: string;
  content: string;
  status: PublicationStatus; // PublicationStatus is an enum, refer to the enum definition for possible values
  createdAt: string; // Represents a DateTime type in string format
  updatedAt: string; // Represents a DateTime type in string format
};
